src/main/java/com/openlattice/chronicle/services/download/ParticipantDataIterable.kt

```diff
@@ -0,0 +1,173 @@
+
+        companion object {
+            private val logger = LoggerFactory.getLogger(ParticipantDataIterable::class.java)
+        }
+
+        override fun hasNext(): Boolean {
+            return hasMorePages
+        }
+
+
+        private fun parseDateTimeValues(values: Set<Any?>, zoneId: ZoneId): Set<Any> {
+            val result: MutableSet<Any> = Sets.newHashSet()
+            values.forEach(Consumer { `val`: Any? ->
+                try {
+                    val parsed = OffsetDateTime.parse(`val`.toString())
+                            .toInstant()
+                            .atZone(zoneId)
+                            .toOffsetDateTime()
+                            .toString()
+                    result.add(parsed)
+                } catch (e: Exception) {
+                    result.add("")
+                }
+            })
+            return result
+        }
+
+        override fun next(): Map<String, Set<Any>> {
+            lock.lock()
+
+            val nextElement: MutableMap<String, Set<Any>> = Maps.newHashMap()
+            try {
+                // if we have finished processing all data in current page, retrieve next
+                if (currentPage == null || finishedCurrentPage) {
+                    currentPage = neighborPageSupplier.get()
+
+                    // if an exception occurs while getting page, stop iterating
+                    if (currentPage == null) {
+                        hasMorePages = false
+                        throw RuntimeException("download failed")
+                    }
+                    currentIndex = 0 // reset index
+                }
+                val neighbors: Map<UUID, List<NeighborEntityDetails>> = currentPage!!.neighbors
+                if (neighbors.isEmpty()) {
+                    hasMorePages = false
+                }
+                val neighborEntityDetails = neighbors[neighborPageSupplier.participantEKID] ?: ImmutableList.of()
+                if (neighborEntityDetails.isNotEmpty()) {
+
+                    // 1: process entity data
+                    val entities = neighborEntityDetails[currentIndex]
+                            .neighborDetails.get()
+                    val zoneId = ZoneId.of(entities
+                            .getOrDefault(EdmConstants.TIMEZONE_FQN, ImmutableSet.of<Any>(OutputConstants.DEFAULT_TIMEZONE))
+                            .iterator()
+                            .next()
+                            .toString()
+                    )
+                    entities.forEach { (key: FullQualifiedName?, `val`: Set<Any?>) ->
+                        if (!neighborPageSupplier.srcPropertiesToExclude.contains(key)) {
+                            val propertyType = neighborPageSupplier.edmCacheManager
+                                    .getPropertyType(key)
+                            val title = OutputConstants.APP_PREFIX + neighborPageSupplier.srcMetadata[propertyType.id]
```

Comment:
title is nullable here. If it shouldn't be nullable use .getValue(...) instead [] index operator which will thrown an NPE if the property type id is not present.